datafusion/datasource-arrow/src/file_format.rs

```diff
@@ -176,14 +184,33 @@ impl FileFormat for ArrowFormat {
 
     async fn create_physical_plan(
         &self,
-        _state: &dyn Session,
+        state: &dyn Session,
         conf: FileScanConfig,
     ) -> Result<Arc<dyn ExecutionPlan>> {
+        let object_store = state.runtime_env().object_store(&conf.object_store_url)?;
+        let object_location = &conf
+            .file_groups
+            .first()
+            .ok_or_else(|| internal_datafusion_err!("No files found in file group"))?
+            .files()
+            .first()
```

Comment:
This makes the assumption that all file groups and all files in them are always the same type - either all plain files or all are stream files.
Is this always the case ?